cmd/clusterctl/examples/vsphere/machines.yaml.template

```diff
@@ -14,29 +14,21 @@ items:
           apiVersion: "vsphereproviderconfig/v1alpha1"
           kind: "VsphereMachineProviderConfig"
           machineSpec:
-            datacenter: ""
-            datastore: ""
-            resourcePool: ""
+            datacenter: "${VSPHERE_DATACENTER}"
+            datastore: "${VSPHERE_DATASTORE}"
+            resourcePool: "${VSPHERE_RESOURCE_POOL}"
+            vmFolder: "${VSPHERE_FOLDER}"
             networks:
-            - networkName: ""
+            - networkName: "${VSPHERE_NETWORK}"
               ipConfig:
                 networkType: static
-                ip: ""
-                netmask: ""
-                gateway: ""
-                dns:
-                - xxxx
-                - yyyy
             numCPUs: 2
             memoryMB: 2048
-            template: ""
+            template: "${VSPHERE_TEMPLATE}"
             disks:
-            - diskLabel: ""
+            - diskLabel: "${VSPHERE_DISK}"
               diskSizeGB: 20
```

Comment:
Hi @andrewsykim,

Please templatize the disk size and do a quick check on the value. It should never be *smaller* than 20GiB as that is the file system size used when creating the images. Cloud-init will automatically *grow* the file system if the disk is larger than 20 GiB, but I'm not sure what happens if the disk is smaller than 20 GiB. Probably nothing good.

Reply:
Should we default the value to 20 though?

Reply:
I'm just saying it shouldn't be allowed to be less than 20. So:

```shell
DISK_SIZE_GIB="${DISK_SIZE_GIB:-20}"
[[ ${DISK_SIZE_GIB} =~ ^[0-9]+$ ]] || DISK_SIZE_GIB=20
[ "${DISK_SIZE_GIB}" -ge "20" ] || DISK_SIZE_GIB=20
```

Reply:
okay fixed this